overlord/devicestate/devicemgr.go

```diff
@@ -989,6 +989,58 @@ func (m *DeviceManager) ensureSeeded() error {
 	return nil
 }
 
+var processAutoImportAssertionsImpl = processAutoImportAssertions
+
+// ensureAutoImportAssertions makes sure that auto import assertions
+// get processed. assertion should be processed while seeding is in progress
+func (m *DeviceManager) ensureAutoImportAssertions() error {
+	if release.OnClassic {
+		return nil
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	if m.earlyDeviceSeed == nil {
+		// we have no seed cached yet, no point to check further
+		return nil
+	}
+
+	var seeded bool
+	if err := m.state.Get("seeded", &seeded); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+	// if system is seeded, stop trying
+	if seeded {
+		return nil
+	}
+
+	// check if we have processed auto import as already
+	var asState string
+	if err := m.state.Get("system-user-assertion", &asState); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+
+	if asState == "done" || asState == "pending" {
```

Comment:
When I initially read this I was a bit puzzled about the possible states. The name "pending" made me assume something would move the state from "pending" to "done" eventually. But AFAIU this is not the case(?) and "pending" is just a final state like "done"(?). If my understanding is correct maybe a small tweak to the name of the state would have helped me, something like `done-but-user-creation-pending` (or "done-bug-pending") or something? (or maybe a small comment?). 

Also as another slightly silly question, is the distinction between "done" and "pending" purely informational? I.e. could we just collapse it into a single state or is the plan to do something about it later (in a followup maybe)?

Reply:
I think the idea is to later try to go from pending to done by taking into account when the device finally gets a serial, but maybe the code as is insufficient to set that up properly and we should have a TODO instead where we set pending and rework this when we can really do the final bits and know what we actually need there

Reply:
Yeah, leaving a TODO here and the slightly ocnfusing pending state out for now sounds good to me.

Reply:
done leaving a TODO